Makefile

```diff
@@ -20,18 +20,21 @@ HEADERS = src/pda.h
 SRC = src/demo.c src/pda.c
 BIN = demo
 CC ?= gcc
-
+AS = *~ src/*~
 
 $(BIN): $(SRC) $(HEADERS)
 	$(CC) $(C_FLAGS) $(CXX_FLAGS) $(SRC) $(HEADERS) -o $(BIN)
 
 debug: $(SRC) $(HEADERS)
 	$(CC) $(C_FLAGS) $(DEBUG) $(SRC) $(HEADERS) -o $(BIN)
 
-clean: $(BIN)
-	rm $(BIN) *~ src/*~
+clean-as: $(AS)
+	rm *~ src/*~
+
+clean: $(BIN) clean-as
+	rm $(BIN)
```

Comment:
This will not solve the problem because it will fail if no file available and before at least we can complete the process with error report